src/m365/spo/commands/tenant/tenant-applicationcustomizer-get.ts

```diff
@@ -0,0 +1,146 @@
+        title: typeof args.options.title !== 'undefined',
+        id: typeof args.options.id !== 'undefined',
+        clientSideComponentId: typeof args.options.clientSideComponentId !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId  [clientSideComponentId]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['title', 'id', 'clientSideComponentId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const spoUrl: string = await spo.getSpoUrl(logger, this.debug);
+      console.log(`${spoUrl}/_api/SP_TenantSettings_Current`);
+      const requestOptions: any = {
+        url: `${spoUrl}/_api/SP_TenantSettings_Current`,
+        headers: {
+          accept: 'application/json;odata=nometadata'
+        }
+      };
+
+      const res: string = await request.get(requestOptions);
+      const json = JSON.parse(res);
+      console.log(json);
+      const appCatalogUrl: string | undefined = json.CorporateCatalogUrl;
+
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      let filter: string = '';
+      if (args.options.title) {
+        filter = `Title eq '${args.options.title}'`;
+      }
+      else if (args.options.id) {
+        filter = `GUID eq '${args.options.id}'`;
+      }
+      else if (args.options.clientSideComponentId) {
+        filter = `TenantWideExtensionComponentId eq '${args.options.clientSideComponentId}'`;
+      }
+
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+      console.log(`${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=${filter}`);
+      const reqOptions: CliRequestOptions = {
+        url: `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=${filter}`,
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      const listItemInstances = await request.get<ListItemInstanceCollection>(reqOptions);
+      console.log(listItemInstances);
+      listItemInstances.value.forEach(v => delete v['ID']);
```

Comment:
Let's do this after we checked if we've got just one result. Otherwise, it might be performing unnecessary work if we end up throwing error anyway.